src/uniq.js

```diff
@@ -18,4 +20,37 @@ var uniqWith = require('./uniqWith');
  *      R.uniq([1, '1']);     //=> [1, '1']
  *      R.uniq([[42], [42]]); //=> [[42]]
  */
-module.exports = uniqWith(equals);
+module.exports = (function() {
+  /**
+   * Uses a native `Set` instance where possible for
+   * removing duplicate items. Items that implement
+   * the fantasy-land Setoid spec fallback to using
+   * `_contains` to support custom equality behaviour.
+   */
+  function uniq(list) {
+    /* global Set */
+    var item,
+        set = new Set(),
+        idx = -1,
+        len = list.length,
+        items = [],
+        uniqs = [];
+
+    while (++idx < len) {
+      item = list[idx];
+      if (item === -0 || typeof item.equals === 'function') {
```

Comment:
@davidchambers: Good catch. The intent was to allow `+0` in the Set, while kicking the `-0` over to the `items` array where they can treated as unique numbers. I've updated it to actually test for `0` and `-0` now. It was co-incidentally still passing the tests because it was falling through and delegating to `R.equals` for comparing `0` and `-0`.

@thefourtheye: I'm not sure I understand. Both paths end up building up the unique items in `uniqs`, regardless of whether `Set` or `_contains` is used.
```
> var x = {}
> uniq([x, 'a', x, 'b', x, 'a', x, 'b', 0])
[ {}, 'a', 'b', 0 ]
```